swan-lake/integration-tools/openapi-tool.md

```diff
@@ -285,6 +287,71 @@ $ bal openapi -i <openapi-contract> --mode client --client-methods <resource|rem
+
+The following is an example of the tool configuration usage in the `Ballerina.toml`.
+
+```ballerina
+[[tool.openapi]]
+id = "client01"
+filePath = "./openapi.yaml"
+targetModule = "delivery"
+options.mode = "client"
+options.statusCodeBinding = true
+options.tags = ["tag1", "tag2"]
+options.operations = ["op1", "op2"]
+options.nullable = true # (default value => false)
+options.license = "./license.txt"
+```
+
+The below tool configuration can be used
+
+| Command option      | Description                                                                                                                                                                                                                                                                                                                                                                     | Mandatory/Optional |
+|----------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
+| `filePath`   | The `filePath` tool option specifies the path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                                                                                                                                                                                                                                                                         | Mandatory          |
+| `id`          | Id for the generation module                                                                                                                                                                                                                                                                                      | Mandatory        |
+| `targetModule`  | The Ballerina files are generated at the given directory name in the Ballerina package.                                                                                                                                                                                                                                                                                     | Optional          |
+| `options.mode`          | Mode type can be client. It generates client stubs for the given OpenAPI contract.                                                                                                                                                                                                                                                                                                 | Optional          |
```

Comment:
```suggestion
| `options.mode`          | The mode type can be `client`. It generates client stubs for the given OpenAPI contract.                                                                                                                                                                                                                                                                                                 | Optional          |
```

Reply:
Shouldn't we mention the all possible mode types? Are they `client` and `service`?

Reply:
Yeah, we can add `service` mode also here. But currently, we only support client generation. Service mode will be supported in future releases. Therefore, I avoid including service mode at this time due to potential confusion for users with documentation. WDYT?

Reply:
Then we better mention it or reword as `can be` is implying that there can be more other types. How about something like below?

The supported mode type is `client`. It generates client stubs for the given OpenAPI contract.

Reply:
This makes clear, that I will use this `The supported mode type is client. It generates client stubs for the given OpenAPI contract.` :)